### PR TITLE
[WIP] Add Option and About to Navigation Drawer

### LINKED ISSUE #407: More Functions in the Navigation Drawer (Options, About, Import)
It is somewhat common to have more buttons at the navigation drawer. 

I suppose that it would be nice to have the following functions directly accessible at the navigation drawer:

* Import
* Export
* Donate
* Options

What do you think?

### PR DESCRIPTION
**Pull request by xie-dongping**
Add option and about to navigation drawer.

Changed the miscellaneous icon to shapes to avoid duplication.

Resolved #407

### CODE REVIEW

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java [hunk 1]
```diff
@@ -350,34 +351,42 @@ public void onClick(DialogInterface arg0, int arg1){
...
+            MiscTabFragment.createAboutDialog(getContext());
+        }
+    }
 
-            View alertView = View.inflate(mActivity, R.layout.link_alert, null);
-            TextView textView = (TextView)alertView.findViewById(R.id.link_alert_message);
-            textView.setText(Html.fromHtml(getString(R.string.about_text)));
-            textView.setMovementMethod(LinkMovementMethod.getInstance());
-            new AlertDialog.Builder(mActivity)
+    public static void createAboutDialog(final Context context) {
```
Comment (helloworld1) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]:
```
A good solution would be adding another util class for displaying the About dialog

Create AboutUtil.java

@PerActivity
public class AboutUtil
    @Inject
    public AboutUtil(Activity activityContext) {
               this.context = activityContext
    }  
    
    public showAboutDialog() {
         // All logic here
    }
  
```

Comment (xie-dongping) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]: Thanks for the suggestion! I'll implement it this way.

**Comment by xie-dongping** [comment]
Hi, it is trivial to implement dependency injection for activity. But somehow I can't get it working for Fragments. I have added a new commit that's not working. Would you mind looking at it for me?

**Comment by helloworld1** [comment]
Don't define it in fragment component., Fragment is not context type. Define it in activity component and fragment component will automatic inherit it.

**Comment by xie-dongping** [comment]
Since the injection still fails, I guess that I'll have to spend some time reading the Dagger2 documentation in details.

**Comment by helloworld1** [comment]
You might forget to call fragmentComponent().inject(this) in onCreate in MiscFragment.java.

**Comment by xie-dongping** [comment]
That's right! Now it works.

**Pull request merged**